packit_service/worker/testing_farm.py

```diff
@@ -67,84 +69,119 @@ def __init__(
 
-    def run_testing_farm(self):
-        for chroot in self.tests_chroots:
-            pipeline_id = str(uuid.uuid4())
-            logger.debug(f"Pipeline id: {pipeline_id}")
-            payload: dict = {
-                "pipeline": {"id": pipeline_id},
-                "api": {"token": self.config.testing_farm_secret},
-            }
+    def report_missing_build_chroot(self, chroot: str):
+        self.status_reporter.report(
+            state="error",
+            description=f"No build defined for the target '{chroot}'.",
+            check_names=PRCheckName.get_testing_farm_check(chroot),
+        )
 
-            logger.debug(f"Payload: {payload}")
+    def run_testing_farm_on_all(self):
+        failed = {}
+        for chroot in self.tests_chroots:
+            result = self.run_testing_farm(chroot)
+            if not result["success"]:
+                failed[chroot] = result.get("details")
+
+        if not failed:
+            return HandlerResults(success=True, details={})
+
+        return HandlerResults(
+            success=False,
+            details={"msg": f"Failed testing farm targets: '{failed.keys()}'."}.update(
+                failed
+            ),
+        )
 
-            stg = "-stg" if self.config.deployment == Deployment.stg else ""
-            copr_repo_name = (
-                f"packit/{self.project.namespace}-{self.project.repo}-"
-                f"{self.event.pr_id}{stg}"
+    def run_testing_farm(self, chroot: str) -> HandlerResults:
+        if chroot not in self.tests_chroots:
+            # Who triggered that?
+            msg = f"Target '{chroot}' not defined for tests but triggered."
+            logger.error(msg)
+            send_to_sentry(PackitConfigException(msg))
+            return HandlerResults(success=False, details={"msg": msg},)
+
+        if chroot not in self.build_chroots:
+            self.report_missing_build_chroot(chroot)
+            return HandlerResults(
+                success=False,
+                details={
+                    "msg": f"Target '{chroot}' not defined for build. "
+                    f"Cannot run tests without build."
+                },
             )
 
-            payload["artifact"] = {
-                "repo-name": self.event.base_repo_name,
-                "repo-namespace": self.event.base_repo_namespace,
-                "copr-repo-name": copr_repo_name,
-                "copr-chroot": chroot,
-                "commit-sha": self.event.commit_sha,
-                "git-url": self.event.project_url,
-                "git-ref": self.base_ref,
-            }
+        check_name = PRCheckName.get_testing_farm_check(chroot)
+        pipeline_id = str(uuid.uuid4())
+        logger.debug(f"Pipeline id: {pipeline_id}")
+        payload: dict = {
+            "pipeline": {"id": pipeline_id},
+            "api": {"token": self.config.testing_farm_secret},
+        }
+
+        logger.debug(f"Payload: {payload}")
```

Comment:
you have one more `logger.debug(payload)` few lines below, do you need both?

Reply:
Hmm, I only moved that code..;) Do we want to debug testing farm token?

Reply:
I don't mind leaking tokens into logs. Whoever can get to the pods logs, can most likely get to secrets too anyway.

I don't understand your note about moving code. (friday, you know)

Reply:
I will remove the payload one.

> moved code

(I haven't written those lines. I've only moved that code from other file, so I am not sure about the reason. Maybe some debugging by Rado from the early stages of testing-farm suppport?)